data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/GetAttestationDataTest.java

```diff
@@ -13,120 +13,81 @@
-  private final JsonProvider jsonProvider = new JsonProvider();
-  private GetAttestationData handler;
-  private AttestationData attestationData =
-      new AttestationData(dataStructureUtil.randomAttestationData());
-
-  @SuppressWarnings("unchecked")
-  final ArgumentCaptor<SafeFuture<String>> resultCaptor = ArgumentCaptor.forClass(SafeFuture.class);
-
-  @BeforeEach
-  public void setup() {
-    handler = new GetAttestationData(provider, jsonProvider);
-  }
+class GetAttestationDataTest extends AbstractMigratedBeaconHandlerTest {
+  private final GetAttestationData handler = new GetAttestationData(validatorDataProvider);
+  private final AttestationData attestationData = dataStructureUtil.randomAttestationData();
 
   @Test
-  void shouldRejectTooFewArguments() throws Exception {
-    badRequestParamsTest(Map.of(), "Please specify both slot and committee_index");
-  }
+  void shouldReturnAttestationDataInformation() throws Exception {
+    request.setQueryParameter(SLOT, "1");
+    request.setQueryParameter(COMMITTEE_INDEX, "1");
 
-  @Test
-  void shouldRejectWithoutSlot() throws Exception {
-    badRequestParamsTest(
-        Map.of("foo", List.of(), "Foo2", List.of()), "'slot' cannot be null or empty.");
-  }
+    when(validatorDataProvider.createAttestationDataAtSlot(UInt64.ONE, 1))
+        .thenReturn(SafeFuture.completedFuture(Optional.of(attestationData)));
 
-  @Test
-  void shouldRejectWithoutCommitteeIndex() throws Exception {
-    badRequestParamsTest(
-        Map.of(SLOT, List.of("1"), "Foo2", List.of()),
-        "'committee_index' cannot be null or empty.");
-  }
+    handler.handleRequest(request);
 
-  @Test
-  void shouldRejectNegativeCommitteeIndex() throws Exception {
-    badRequestParamsTest(
-        Map.of(SLOT, List.of("1"), COMMITTEE_INDEX, List.of("-1")),
-        "'committee_index' needs to be greater than or equal to 0.");
+    assertThat(request.getResponseCode()).isEqualTo(SC_OK);
+    assertThat(request.getResponseBody()).isEqualTo(attestationData);
   }
 
   @Test
   void shouldReturnNoContentIfNotReady() throws Exception {
-    Map<String, List<String>> params = Map.of(SLOT, List.of("1"), COMMITTEE_INDEX, List.of("1"));
+    request.setQueryParameter(SLOT, "1");
+    request.setQueryParameter(COMMITTEE_INDEX, "1");
 
-    when(context.queryParamMap()).thenReturn(params);
-    when(provider.createAttestationDataAtSlot(UInt64.ONE, 1))
+    when(validatorDataProvider.createAttestationDataAtSlot(UInt64.ONE, 1))
         .thenReturn(SafeFuture.failedFuture(new ChainDataUnavailableException()));
-    handler.handle(context);
 
-    verify(context).future(resultCaptor.capture());
-    final SafeFuture<String> result = resultCaptor.getValue();
-    assertThat(result).isCompletedExceptionally();
-    assertThatThrownBy(result::join).hasRootCauseInstanceOf(ChainDataUnavailableException.class);
+    handler.handleRequest(request);
+
+    assertThat(request.getResponseBody()).isNull();
```

Comment:
whats the response code here? that's more important than the body being empty

Reply:
It seems to be -1, does that mean I must be missing something. That isn't a valid option

Reply:
we'll want to get a better understanding there for sure...